fractal_server/app/routes/auth.py

```diff
@@ -53,17 +70,45 @@
 )
 
 
-@router_auth.get("/whoami", response_model=UserRead)
-async def whoami(
-    user: User = Depends(current_active_user),
+@router_auth.patch("/current-user/", response_model=UserRead)
+async def patch_current_user(
+    user_update: UserUpdateStrict,
+    current_user: User = Depends(current_active_user),
+    user_manager: UserManager = Depends(get_user_manager),
 ):
+
+    try:
+        user = await user_manager.update(
+            UserUpdate(**user_update.dict(exclude_unset=True)),
+            current_user,
+            safe=True,
+        )
+    except exceptions.InvalidPasswordException as e:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail={
+                "code": ErrorCode.UPDATE_USER_INVALID_PASSWORD,
+                "reason": e.reason,
+            },
+        )
+    except exceptions.UserAlreadyExists:
```

Comment:
I don't think we can ever enter this branch, and in that case I would not include it.

That's only my first guess, but it's easy to check: if we can have test coverage for these lines, then let's keep them.